--- a/src/nm-l2tp-service.c
+++ b/src/nm-l2tp-service.c
@@ -678,9 +678,13 @@
 
 	write_config_option (ipsec_fd, 		"  authby=secret\n"
 						"  keyingtries=0\n"
-						"  left=%%defaultroute\n"
-						"  leftprotoport=udp/l2tp\n"
-						"  rightprotoport=udp/l2tp\n");
+						"  left=%%defaultroute\n");
+
+	if (is_port_free (1701)) {
+		write_config_option (ipsec_fd, "  leftprotoport=udp/l2tp\n");
+	}
+	write_config_option (ipsec_fd, "  rightprotoport=udp/l2tp\n");
+
 	value = nm_setting_vpn_get_data_item (s_vpn, NM_L2TP_KEY_IPSEC_GROUP_NAME);
 	if (value) {
 		if (priv->is_libreswan) {
@@ -707,6 +711,8 @@
 		} else {
 			write_config_option (ipsec_fd, "  rightid=%s\n", value);
 		}
+	} else {
+		write_config_option (ipsec_fd, "  rightid=%%any\n");
 	}
 
 	if (priv->is_libreswan) {
@@ -991,30 +997,6 @@
 		return nm_l2tp_ipsec_error(error, "Could not write /etc/ipsec.secrets file.");
 	}
 
-	value = nm_setting_vpn_get_data_item (s_vpn, NM_L2TP_KEY_IPSEC_GROUP_NAME);
-	if (value) {
-		if(inet_pton(AF_INET, value, &naddr)) {
-			fprintf(fp, "%s ", "%any");
-		} else {
-			/* @ prefix prevents lefttid being resolved to an IP address */
-			fprintf(fp, "@%s ", value?value:"%any");					
-		}
-	} else {
-		fprintf(fp, "%s ", value?value:"%any");
-	}
-	
-	value = nm_setting_vpn_get_data_item (s_vpn, NM_L2TP_KEY_IPSEC_GATEWAY_ID);
-	if (value) {
-		if(inet_pton(AF_INET, value, &naddr)) {
-			fprintf(fp, "%s ", "%any");
-		} else {
-			/* @ prefix prevents rightid being resolved to an IP address */
-			fprintf(fp, "@%s ", value?value:"%any");					
-		}
-	} else {
-		fprintf(fp, "%s ", value?value:"%any");
-	}	
-
 	value = nm_setting_vpn_get_data_item (s_vpn, NM_L2TP_KEY_IPSEC_PSK);
 	if(!value)value="";
 	fprintf(fp, ": PSK \"%s\"\n",value);
